### PR TITLE
feat: add upvote log

### PR DESCRIPTION
**Pull request by catngh**
**What does this PR do?**

- Add upvote log to get time left for each upvote source
- `nft/new-listed` now returns null and code 200 if `record not found`
- New `users/upvote-streak` response
```
{
    "data": {
        "discord_id": "319132138849173505",
        "minutes_until_reset": 780,
        "minutes_until_reset_topgg": 780,
        "minutes_until_reset_discordbotlist": 780,
        "streak_count": 7,
        "total_count": 22,
        "last_streak_time": "2022-09-12T10:33:00Z"
    }
}
```

### CODE REVIEW

File: pkg/entities/user.go [hunk 1]
```diff
@@ -96,19 +97,36 @@ func (e *Entity) GetUserCurrentUpvoteStreak(discordID string) (*response.GetUser
...
+	if err != nil {
+		e.log.Info("[e.GetUserCurrentUpvoteStreak] user first time upvote")
+	}
+	for _, log := range upvoteLogs {
+		expireTime = log.LatestUpvoteTime.Add(time.Hour * 13)
+		switch log.Source {
+		case "topgg":
+			topggTime = util.MinuteLeftUntil(currTime, expireTime)
+		case "discordbotlist":
+			dcBotTime = util.MinuteLeftUntil(currTime, expireTime)
```
Comment (trkhoi) [inline pkg/entities/user.go:118]: what happen with `default` case

File: pkg/handler/nfts.go [hunk 2]
```diff
@@ -384,6 +384,11 @@ func (h *Handler) GetNewListedNFTCollection(c *gin.Context) {
 
 	data, err := h.entities.GetNewListedNFTCollection(interval, page, size)
 	if err != nil {
+		if strings.Contains(err.Error(), "record not found") {
```
Comment (trkhoi) [inline pkg/handler/nfts.go:387]: pls provide bot image of `nft recent`

File: pkg/handler/webhook.go [hunk 3]
```diff
@@ -190,7 +190,7 @@ func (h *Handler) WebhookUpvoteTopGG(c *gin.Context) {
 		return
 	}
 
-	err = h.entities.WebhookUpvoteStreak(req.UserID)
+	err = h.entities.WebhookUpvoteStreak(req.UserID, "topgg")
```
Comment (trkhoi) [inline pkg/handler/webhook.go:193]: `topgg` and `discordbotlist` should add in `constant`

**Review by trkhoi — CHANGES_REQUESTED** [review]

File: pkg/entities/user.go (hunk 1 above)
Comment (catngh) [reply pkg/entities/user.go:118]: i init them = 0

**Review by trkhoi — APPROVED** [review]

**Review by nguyennh4522 — APPROVED** [review]

**Pull request merged**